### DISCUSSION TITLE
How to run different build commands base on git branch?

### DISCUSSION THREAD

**Discussion by developer-zmo** [vercel/community#3155] [Help]
### Summary

I have two branches with "main" and "dev", With dev branch I want to build a app for test workers  with as build commad as  "yarn build", and I will build for custom with "main" branch with "yarn build:prod"

### Steps to Reproduce

I have two branches with "main" and "dev", With dev branch I want to build a app for test workers  with as build commad as  "yarn build", and I will build for custom with "main" branch with "yarn build:prod"

I tried add a "vercel.sh" file at root directory, and its like 
```ts
#!/bin/bash
 
if [[ $VERCEL_GIT_COMMIT_REF == "main"  ]] ; then 
  echo "This is our main branch"
  yarn build:prod
else 
  echo "This is not our main branch"
  yarn build
fi
```

and then I overwirte the build command with "bash ./vercel.sh", like this: 
![image](https://github.com/vercel/community/assets/128699008/b7b5526b-da28-461f-b9a7-39b624a4f739)
 
 it runs my build script but caucht an error and has no detail 
![image](https://github.com/vercel/community/assets/128699008/7d04c47a-6fc8-436f-b07f-ea33191d06cd)

Then I try to find a way from vercel.json configuration file, but I still did not find anything, please teach me what should I do, thanks so much

**Comment by gczh** [comment]
Have you tried doing `sh vercel.sh` instead of `bash ./vercel.sh`?

Here's a reference: https://www.storyblok.com/tp/different-build-commands-branch-name-vercel
